pkg/action/executor/rescheduler.go

```diff
@@ -85,45 +85,79 @@ func (r *ReScheduler) checkActionItem(action *proto.ActionItemDTO) error {
 	return nil
 }
 
-// get k8s.nodeName of the node
-// TODO: get node info via EntityProperty
+// get k8s.node of the new hosting node
 func (r *ReScheduler) getNode(action *proto.ActionItemDTO) (*api.Node, error) {
 	hostSE := action.GetNewSE()
 
 	var err error = nil
 	var node *api.Node = nil
 
-	if r.stitchType == stitching.UUID {
-		node, err = util.GetNodebyUUID(r.kubeClient, hostSE.GetId())
-	} else if r.stitchType == stitching.IP {
-		var machineIPs []string
-		if hostSE.GetEntityType() == proto.EntityDTO_VIRTUAL_MACHINE {
-			vmData := hostSE.GetVirtualMachineData()
-			if vmData == nil {
-				err := fmt.Errorf("Missing virtualMachineData[%v] in targetSE.", hostSE.GetDisplayName())
-				glog.Error(err.Error())
-				return nil, err
-			}
-			machineIPs = vmData.GetIpAddress()
+	//0. check entity type
+	etype := hostSE.GetEntityType()
+	if etype != proto.EntityDTO_VIRTUAL_MACHINE && etype != proto.EntityDTO_PHYSICAL_MACHINE {
+		err = fmt.Errorf("The target entity(%v) for move destiantion is neither a VM nor a PM.", etype)
+		glog.Error(err.Error())
+		return nil, err
+	}
+
+	//1. get node from properties
```

Comment:
Here, we try four ways to get node:
1. from entity properties
2. from entity displayName
3. from entity uuid
4. from vm ip list

From my understand, the entity display name is set as the k8s node name. So, we can simply use the display name to get the k8s node. Why do we need the four ways to get it? Is it because some name truncating situation?

Reply:
yea, because of name truncating, displayname is not very reliable.